siliconcompiler/core.py

```diff
@@ -120,22 +120,9 @@ def __init__(self, design, loglevel=None):
             'checklists': []
         }
 
-        # Custom error handlers used to provide warnings when invalid characters
-        # are encountered in a file for a given encoding. The names
-        # 'replace_with_warning' and 'ignore_with_warning' are supplied to
-        # open() via the 'errors' kwarg.
-
-        # Warning message/behavior for invalid characters while running tool
-        def display_error_handler(e):
-            self.logger.warning('Invalid character in tool output, displaying as �')
-            return codecs.replace_errors(e)
-        codecs.register_error('replace_with_warning', display_error_handler)
-
-        # Warning message/behavior for invalid characters while processing log
-        def log_error_handler(e):
-            self.logger.warning('Ignoring invalid character found while reading log')
-            return codecs.ignore_errors(e)
-        codecs.register_error('ignore_with_warning', log_error_handler)
+        # Initialize custom error handling for codecs. This has to be called
+        # by each spawned (as opposed to forked) subprocess
+        self._init_codecs()
```

Comment:
Could you move this up to above the `init_logger` (on line 80) to mirror the setup in `_run_task`